### PR TITLE
Add Azure Batch DataPlane SDK version 10

### PR DESCRIPTION
**Pull request by matthchr**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

For SDK team:
1. The REST API PR for this REST API is here: https://github.com/Azure/azure-rest-api-specs/pull/4795
2. This change *does* introduce breaking changes
3. Test coverage was added where appropriate (most model changes are covered by existing reflection based testing)

### CODE REVIEW

**Comment by matthchr** [comment]
SDK team folks, please allow some time for my coworkers @xingwu1 and @bgklein to review prior to merging.

**Comment by matthchr** [comment]
This also deletes the extra packages.config file which was causing #4880

**Comment by xingwu1** [comment]
Looks good to me.

File: src/SDKs/Batch/DataPlane/changelog.md [hunk 1]
```diff
@@ -1,8 +1,41 @@
...
+  now dynamically assign a Virtual Network to each node the job's tasks run on. The specific Virtual Network to join the nodes to is specified in 
+  the new `JobNetworkConfiguration` property on `CloudJob` and `JobSpecification`. 
+  - **Note**: This feature is in public preview. It is disabled for all Batch accounts except for those 
+- The maximum lifetime of a task is now 180 days (previously it was 7).
+- Added support on Windows pools for creating users with a specific login mode (either `Batch` or `Interactive`) via `WindowsUserConfiguration.LoginMode`.
+- The default task retention time for all tasks is now 7 days, previously it was infinite.
+- `ExponentialRetry` supports a backoff cap, via the `MaxBackoff` property.
+
+### Bug fixes
+- The built in retry policies `ExponentialRetry` and `LinearRetry` now correctly retry on HTTP status code `429` and honor the `retry-after` header.
```
Comment (bgklein) [inline src/SDKs/Batch/DataPlane/changelog.md]: Is this really a 'bug'

**Comment by bgklein** [comment]
LGTM as well

File: src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj [hunk 2]
```diff
@@ -10,7 +10,7 @@
     <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
     <PackageId>Microsoft.Azure.Batch</PackageId>
     <Description>This client library provides access to the Microsoft Azure Batch service.</Description>
-    <Version>9.0.1</Version>
+    <Version>10.0.0</Version>
```
Comment (shahabhijeet) [inline src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj]: @matthchr why a major version bump?
Can you also release notes?
Is the major version due to breaking changes or underlying API change?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj (hunk 2 above)
Comment (matthchr) [reply src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj]: I did release notes, they're included in the PR.
There are breaking changes both in the SDK and in the underlying API, which is why it's a major version bump. The release notes outlines what changed and what breaks there were.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**